### ISSUE TITLE
[docs] rh-navigation-secondary current page indicator

### ISSUE THREAD

**Issue by zeroedin** [RedHat-UX/red-hat-design-system#1191]
From design system chat: 

> Wes Carr,
> secondary nav question
> hey hey. i'm not seeing the current page indicator working on live pages, but it's here in the docs. is there a timeline for it being updated or is there a manual way to toggle it on? https://ux.redhat.com/elements/navigation-secondary/guidelines/#current-page-indicator

Guidelines on https://ux.redhat.com/elements/navigation-secondary/guidelines/#current-page-indicator state we should have a current page indicator.  

### Discovery needed:
Should this responsibility fall on the component or implementation?

A `current` attribute or class could be added to the dropdown `<a>` to manipulate lightdom styles.  Is it possible that each implementation maybe have a different way of marking that current page? Such as using [attributes] vs .classes vs naming conventions such as `current` vs `current-page`

lightdom css example:

```css
:is(rh-secondary-nav-dropdown, rh-navigation-secondary-dropdown) a[current],
:is(rh-secondary-nav-dropdown, rh-navigation-secondary-dropdown) a.current {
    border-block-start-color: var(--rh-color-brand-red-on-light, #ee0000);
}

:is(rh-secondary-nav-dropdown, rh-navigation-secondary-dropdown)[color-palette="dark"] a[current],
:is(rh-secondary-nav-dropdown, rh-navigation-secondary-dropdown)[color-palette="dark"] a.current {
    border-block-start-color: var(--rh-color-brand-red-on-dark, #ff442b);
}
```

Implementors can currently use the example above to accomplish this requirement of the current page indicator

Dropdown links are "possible" index pages that never get navigated to unless the navigation fails to upgrade and the dropdown takes over from the default link function, so they themselves don't always represent a navigable page that will be "current".  However any number of sub pages in their dropdowns maybe current, so the CMS or implementation would have to determine when to make the dropdown show this indication.  Also its possible that a link exists under more then one dropdown in a navigation menu.

**Comment by brianferry** [comment]
This seems on the level of the history api with tabs / accordion where it will need to fall back onto the implementation of the component on the pages.  We could provide examples / patterns for how to manage the highlighting based on a url / title

**Comment by wesleymiles** [comment]
hey,

re: whether to have the current page indicator on nav items with a dropdown. i think the answer to that is yes based on the [specs on ux dot](https://ux.redhat.com/elements/navigation-secondary/guidelines/#current-page-indicator). 

re: how it's done in drupal? i have no idea. would we want to loop in a drupal dev to help collab? 
![Screenshot 36](https://github.com/RedHat-UX/red-hat-design-system/assets/613815/b374cfe4-7d33-4bcb-9603-75f226f7afa5)

we _do_ want to avoid showing 2 current page indicators if an item is within multiple panels. but that would be rare going by navigation ux standards. good catch to consider that logic as a safeguard. 

basically, i'm wanting it to work like the screenshot.

**Comment by zeroedin** [comment]
It has been decided that this becomes integration documentation that needs to be added to ux.redhat.com.  

/cc @marionnegp @coreyvickery this is something we can talk about.  Will need some word smithing and figuring out where this lives on the site.

**Comment by markcaron** [comment]
@zeroedin IMO, it seems like something that would live under the `Code` section for Navigation Secondary. This is specific to integration and not a separate "pattern".

Word-smithing-wise, perhaps under `Installation` with a subheading of `Setting 'current' page` or something similar.

**Comment by nikkimk** [comment]
For accessibility, the link that goes to the current page should have `aria-current="page"`, so rather than supporting a class that only benefits sighted users, CSS should be based on `[aria-current="page"]`.

**Comment by zeroedin** [comment]
> For accessibility, the link that goes to the current page should have `aria-current="page"`, so rather than supporting a class that only benefits sighted users, CSS should be based on `[aria-current="page"]`.

Back tracking the approach to use `aria-current`.  Given a base selector we can move this into lightdom/shadowdom CSS styles then look for that selector as a child of the dropdown in order to style the active dropdown. Implementation will still need to be responsible for adding the `aria-current` attribute on to the correct links.
